lib/src/rules/eol_last.dart

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+
+import '../analyzer.dart';
+
+const _desc = r'Put an end of line for non-empty files.';
+
+const _details = r'''
+
+**DO** put an end of line for non-empty files.
+
+**BAD:**
+```dart
+a {
+}
+```
+
+**GOOD:**
+```dart
+b {
+}
+
+''';
+
+class EolLast extends LintRule implements NodeLintRule {
+  EolLast()
+      : super(
+            name: 'eol_last',
+            description: _desc,
+            details: _details,
+            group: Group.style);
+
+  @override
+  void registerNodeProcessors(
+      NodeLintRegistry registry, LinterContext context) {
+    var visitor = _Visitor(this, context);
+    registry.addCompilationUnit(this, visitor);
+  }
+}
+
+class _Visitor extends SimpleAstVisitor<void> {
+  final LintRule rule;
+  final LinterContext context;
+
+  _Visitor(this.rule, this.context);
+
+  @override
+  void visitCompilationUnit(CompilationUnit node) {
+    var endToken = node.endToken;
+    var previous = endToken.previous;
+    if (previous == null) {
+      return;
+    }
+    if (previous.end == endToken.offset ||
```

Comment:
I'm guessing that this is to ensure that the lint doesn't trigger on empty files. I think this will work (I think it will always be true for an empty file and false for a file ending with either whitespace or a comment), but I was thinking of something like
```
var content = context.currentUnit.content;
if (content.isNotEmpty && !content.endsWithNewLine) ...
```
That seems more explicit to me. It took me a while to convince myself that I fully understood the semantics of the comparison.

Reply:
Thanks for guiding new contributors.